src/main/java/com/example/spring/service/user/UserServiceImpl.java

```diff
@@ -34,6 +36,8 @@ public UserResponseDto register(
                 .orElseThrow(() -> new EntityNotFoundException("Role USER not found"));
         user.setRoles(Set.of(userRole));
         user.setPassword(passwordEncoder.encode(request.getPassword()));
+        cartService.createShoppingCart(user);

```

Comment:
```suggestion
        userRepository.save(user);
        cartService.createShoppingCart(user);
```

In such order you won't get LazyInitExcep in getUsersCart(Long userId); so you can remove Transactional

Reply:
order doesn`t matter, lazyinitexception because of book.title in CartItemDto